crates/sui-protocol-config-macros/src/lib.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+extern crate proc_macro;
+
+use proc_macro::TokenStream;
+use quote::{format_ident, quote};
+use syn::{parse_macro_input, Data, DeriveInput, Fields, Type};
+
+/// This macro derives getters for each field in ProtocolConfig at a specific version
```

Comment:
I'll leave it for @mystenmark to comment on this approach, to me it seems a bit overkill but maybe fine since you already did it? I was thinking that instead of a proc macro which makes these for every Option field, you'd just change the accessor methods you added in the other file for your new fields to not call `expect` on the Option

Reply:
Might seem overkill for this specific PR, but I expect more scenarios where we need to gate things this way. And if not needed, folks can simply not call the functions.